fireplace-swing/src/main/java/io/github/bric3/fireplace/flamegraph/FlamegraphView.java

```diff
@@ -865,12 +868,24 @@ public void mouseClicked(MouseEvent e) {
 
             if (e.getClickCount() == 2) {
                 // find zoom target then do an animated transition
-                canvas.getFlamegraphRenderEngine().flatMap(fgp -> fgp.calculateZoomTargetForFrameAt(
-                        (Graphics2D) canvas.getGraphics(),
-                        canvas.getBounds(),
-                        canvas.getVisibleRect(),
-                        latestMouseLocation
-                )).ifPresent(zoomTarget -> zoom(canvas, zoomTarget));
+                canvas.getFlamegraphRenderEngine().flatMap(fgp -> {
+                    if (this.zoomHorizontalOnly) {
```

Comment:
**chore:** Could be replaced by something like `FlamegraphView.from(canvas).orElse(false)`.